cms/djangoapps/contentstore/tests/test_courseware_index.py

```diff
@@ -58,6 +58,14 @@
 
 class NonIndexableXBlock(XBlock):
     """XBlock for testing indexibility"""
+    def index_dictionary(self):
+        # The base class now has a default index_dictionary.
+        # In order to be non-indexable, we have to pretend that
```

Comment:
I guess my main question would be, is this case even applicable any longer?
If all blocks are indexable, should we even be testing for non-indexable?

Reply:
Yeah, it seems to me that the test is not that relevant anymore, but there is still a branch in the code that skips out on indexing if there's nothing to index. So, if we care about covering all our bases (or branches), might as well leave the test in?

Reply:
What does that branch look like? Is it this or another one?

```py
# if it's not indexable and it does not have children, then ignore
if not item_index_dictionary and not item.has_children:
    return
```

In that case, it seems we'd be better served to have this method return `None` or an empty dict, or some such.
If there are already tests for that (a `None`/empty return value), great! If not, that seems more like what we should be testing here, instead of throwing an exception that cannot occur in our application (only in tests).